Assets/Scripts/Town/PlayerInteractions.cs

```diff
@@ -20,27 +22,48 @@ void Update()
                     useLayerMask = true,
                     useTriggers = true
                 },
-                result);
+                newInteractables);
 
-            var newInteractables = new HashSet<IInteractable>(result.Select(x => x.GetComponent<IInteractable>()));
+            float closestInteractibleDistance = float.PositiveInfinity;
+            Collider2D closestInteractable = null;
 
             foreach (var interactable in newInteractables)
             {
-                if (!_oldInteractables.Contains(interactable))
+                // Maybe later change so you can choose which collider is used.
+                var playerInteractionCollider = GetComponent<Collider2D>();
+                var distance = Physics2D.Distance(playerInteractionCollider, interactable).distance;
+                if (distance < closestInteractibleDistance)
                 {
-                    interactable.EnteredInteractionRange();
+                    closestInteractable = interactable;
+                    closestInteractibleDistance = distance;
                 }
             }
 
-            foreach (var interactable in _oldInteractables)
+            if (closestInteractable != null)
             {
-                if (!newInteractables.Contains(interactable))
+                if (_currentInteractable == null)
                 {
-                    interactable.LeftInteractionRange();
+                    closestInteractable.GetComponent<IInteractable>().EnteredInteractionRange();
+                    _currentInteractable = closestInteractable;
                 }
+                else if (_currentInteractable != closestInteractable)
+                {
+                    _currentInteractable.GetComponent<IInteractable>().LeftInteractionRange();
+                    closestInteractable.GetComponent<IInteractable>().EnteredInteractionRange();
+                    _currentInteractable = closestInteractable;
+                }
+            } 
+            else if (_currentInteractable != null)
+            {
+                _currentInteractable.GetComponent<IInteractable>().LeftInteractionRange();
+                _currentInteractable = null;
             }
 
-            _oldInteractables = newInteractables;
+            
+            if (_currentInteractable != null && CInput.InputActions.Town.Interact.triggered)
```

Comment:
Przeniósłbym obsługę input actions do osobnej funkcji.
Dla przykładu:
```
void awake(){
Cinput.InputActions.Town.interact.performed += Interaction;
}

void Interation(CallbackContext whatever){
if(currentInteractable != null){
 currentInteractable.Interact();
}
}
```

Reply:
Czy callbacki wywoływane są przed czy po Update()? Jakie są zalety użycia callbacków nad pollowaniem inputu w tym przypadku?

Reply:
> The callbacks are triggered as part of input system updates (see [Update()](https://docs.unity3d.com/Packages/com.unity.inputsystem@1.0/api/UnityEngine.InputSystem.InputSystem.html#UnityEngine_InputSystem_InputSystem_Update)), i.e. they happen before the respective MonoBehaviour.Update or MonoBehaviour.FixedUpdate methods get executed (depending on which [updateMode](https://docs.unity3d.com/Packages/com.unity.inputsystem@1.0/api/UnityEngine.InputSystem.InputSettings.html#UnityEngine_InputSystem_InputSettings_updateMode) the system is set to).

> Note that actions are not generally frame-based. What this means is that an action will observe any value change on its connected controls, even if the control changes value multiple times in the same frame. In practice, this means that, for example, no button press will get missed.

Ogólnie nie jest to ważna zmiana, w tym przypadku najbardziej liczy się to że wszystkie input'y tego typu chcemy rejestrować przez właśnie callbacki, więc spowoduje to zachowanie tego samego stylu w całym projekcie. Jest to też ogólnie preferowana metoda ich zczytywania.